antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/atn/ATNDeserializer.kt

```diff
@@ -174,16 +174,14 @@ public open class ATNDeserializer(deserializationOptions: ATNDeserializationOpti
     for (state in atn.states) {
       if (state is BlockStartState) {
         // We need to know the end state to set its start state
-        if (state.endState == null) {
-          throw IllegalStateException()
-        }
+        val endState = state.endState ?: throw IllegalStateException()
```

Comment:
Couldn't you merge this and the following check into:
```
val startStateOfEndState = state.endState?.startState ?: throw IllegalStateException()
```

Reply:
@ftomassetti Nope, because that code desugars to:
```kotlin
val startState = state.endState?.startState

if (startState == null) {
  throw IllegalStateException()
}
```
Which is the opposite of the condition we need to check for.

Reply:
Right, sorry